src/components/Wallet.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import AddSigners from './AddSigners';
+import Signers from './Signers';
+import { Device } from '../types'
+import { AppState } from '../store';
+import { getWallets, selectCandidateDevicesForActiveWallet, selectActiveWallet } from '../store/wallet';
+import { Spinner } from 'reactstrap';
+import api from '../api'
+
+interface StateProps {
+  candidateDevices: Device[];
+  deviceError: Error | null;
+  activeWallet: AppState['wallet']['activeWallet'];
+}
+
+interface DispatchProps {
+  getWallets: typeof getWallets;
+}
+
+interface State {
+  deviceBeingAdded: Device | null;
+}
+
+type Props = StateProps & DispatchProps;
+
+class Wallet extends React.Component<Props, State> {
+  state: State = {
+    deviceBeingAdded: null,
+  }
+  private addSigner = async (device: Device) => {
+    if (this.props.activeWallet) {
+      this.setState({ deviceBeingAdded: device })
+      await api.addSigner({
+        wallet_name: this.props.activeWallet.name,
+        signer_name: device.type,
+        device_id: device.fingerprint,
+      })
+      // FIXME: state.wallet.activeWallet goes stale here
+      await this.props.getWallets();
+    }
+  }
+  render() {
+    const { deviceBeingAdded } = this.state;
+    const { candidateDevices, activeWallet, deviceError } = this.props;
+    if (activeWallet === null) {
+      return <div>no active wallet</div>
+    }
+    const { signers } = activeWallet;
+
+    let signersComponent = null;
+    if (signers.length) {
+      signersComponent = (
+        <div>
+          <h3 className='text-center'>Signers</h3>
+          <Signers signers={signers} />
+        </div>
+      )
+    }
+
+    // "Add Signers" section
+    let addSigners = null;
+    if (!activeWallet.ready) {
+      addSigners = (
+        <div>
+          <h3 className='text-center'>Add {activeWallet.n - activeWallet.signers.length} More Signers</h3>
+          <AddSigners devices={candidateDevices} deviceError={deviceError}  deviceBeingAdded={deviceBeingAdded} addSigner={this.addSigner}/>
+        </div>
+      )
+    }
+    return (
+      <div>
+        <h2 className='text-center'>{ activeWallet.name } ({activeWallet.m}/{activeWallet.n})</h2>
+        {activeWallet.ready && 
+          <div>Confirmed Balance: {activeWallet.balances.confirmed}</div>
+        }
+        {activeWallet.balances.unconfirmed > 0 &&
+          <div>Unconfirmed Balance: {activeWallet.balances.unconfirmed}</div>}
+        {signersComponent}
+        {addSigners}
```

Comment:
Might be good to stick with one style of conditional rendering for consistency's sake. Here both variable assignment in conditional, and in-jsx checking is done. `addSigners` and `signersComponent` could both be done with in-jsx conditionals.